### PR TITLE
Sosdba oracle

### PR DESCRIPTION
**Pull request by mricois**
V2.2 - add tip for alert.log , delete dead links, and change rollup to latest which are necessary for oracle metrics.

### CODE REVIEW

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  signal:
+    metric: tnslsnr
+
+rules:
+  critical:
+    threshold: 1
+    comparator: "<"
+    lasting_duration: "5m"
+    description: "is not started"
+tip: "the listener process is not started, means that application can not connect to the database"
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/01-listener.yaml:18]: while you aggregate on `hostname` I would say:
```suggestion
tip: "the listener process is not started, means that application can not connect to the database host"
```

File: modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+name: "Database status"
+id: dbisdown
+
+transformation: true
+aggregation: ".mean(by=['dbname','dbtype','hostname'])"
+
+signals:
+  signal:
+    metric: oracledb_up
+    rollup: latest
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml:11]: Can you explain to me what is for ? if you have "concurrent" datapoint for this MTS : `oracledb_up[dbname,dbtype,hostname']` perhaps it is better to simply not aggregate at all here (or add a dimension a source level if it miss something to differenciate them)

File: modules/prometheus-exporter_oracle/conf/05-alertlog.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module: oracle
+name: "Alert.log errors count"
+id: AlertLogError
+
+transformation: true
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/05-alertlog.yaml:5]:
```suggestion
transformation: true
aggregation: true
``` 

here you do not force aggregation at all, can you not do the same for other detectors ?

File: modules/prometheus-exporter_oracle/conf/09-stby.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+transformation: true
+aggregation: ".mean(by=['dbname','dbtype','hostname'])"
+
+signals:
+  signal:
+    metric: oracledb_STBY_Replication_count 
+    rollup: latest
+
+rules:
+  critical:
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/09-stby.yaml:14]: I would say `major` here, this is not related to direct/immediate availability problem

File: modules/prometheus-exporter_oracle/conf/11-rmanincr.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+transformation: true
+aggregation: ".mean(by=['dbname','dbtype','hostname'])"
+
+signals:
+  signal:
+    metric: oracledb_Oracle_RMAN_Incr_value
+    rollup: latest
+
+rules:
+  critical:
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/11-rmanincr.yaml:14]: minor ?

File: modules/prometheus-exporter_oracle/conf/12-rmanarch.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+transformation: true
+aggregation: ".mean(by=['dbname','dbtype','hostname'])"
+
+signals:
+  signal:
+    metric: oracledb_Oracle_RMAN_Arch_value
+    rollup: latest
+
+rules:
+  critical:
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/12-rmanarch.yaml:14]: minor ?

File: modules/prometheus-exporter_oracle/conf/13-userexpire.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+transformation: true
+aggregation: ".mean(by=['dbname','dbtype','hostname'])"
+
+signals:
+  signal:
+    metric: oracledb_User_pass_expiration_V2_count
+    rollup: latest
+
+rules:
+  critical:
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/13-userexpire.yaml:13]: warning

File: modules/prometheus-exporter_oracle/conf/10-dbexport.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+module: oracle
+name: "database last export"
+id: OracleDB_export
+
+transformation: true
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/10-dbexport.yaml:5]:
```suggestion
transformation: true
aggregation: true
```

File: modules/prometheus-exporter_oracle/conf/14-tablespacecdb.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+module: oracle
+name: "tablespace usage on container database"
+id: Tablespace_CDB
+
+transformation: true
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/14-tablespacecdb.yaml:5]:
```suggestion
transformation: true
aggregation: true
```

File: modules/prometheus-exporter_oracle/conf/16-tablespace.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+module: oracle
+name: "tablespace usage on non-CDB database"
+id: Tablespace_Single
+
+transformation: true
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/16-tablespace.yaml:5]:
```suggestion
transformation: true
aggregation: true
```

File: modules/prometheus-exporter_oracle/conf/17-dbvagent.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+module: oracle
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/17-dbvagent.yaml:1]: Perhaps it is only my lack of knowledge about oracle but perhaps a tip could be useful for this detector.
For instance, here I cannot say if critical severity is relevant or not.

File: modules/prometheus-exporter_oracle/conf/18-dbvnet.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+module: oracle
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/18-dbvnet.yaml:1]: same here for the tip, I don't know what is the job of this process. if this is obvious for someone who know oracle you can ignore

File: modules/prometheus-exporter_oracle/conf/19-dbvctl.yaml [hunk 13]
```diff
@@ -0,0 +1,19 @@
+module: oracle
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/19-dbvctl.yaml:1]: again need a tip?

File: modules/prometheus-exporter_oracle/conf/readme.yaml [hunk 14]
```diff
@@ -0,0 +1,2 @@
+documentations:
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/readme.yaml:1]: link to prometheus exporter and perhaps also some oracle database monitoring ref

File: modules/prometheus-exporter_oracle/conf/readme.yaml [hunk 15]
```diff
@@ -0,0 +1,2 @@
...
+source_doc:
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracle/conf/readme.yaml:2]: this module will not be usable while there is no doc here to explain how to deploy the metrics collection

**Review by xp-1000 — COMMENTED** [review]
Congrats it seems overall good.

I left some questions / suggestions and comments, please answer to them and resolve them when you estimate it is done (or to ignore).

Here are some global remarks:
- should you name the module `oracledb` instead of `oracle` : first it will match with the prometheus exporter name and its metrics, second it will allow us to have another module for another oracle product not related to database
- I am pretty sure there a too many critical severity and at least some detectors deserve to be decreased (backup and replication at least)
- this module requires a doc to make it usable for everybody
- I will need more info about the rollup requirement, especially why exactly and if we can avoid that which is not desirable
- I am sure some detectors here should use an aggregation a "host" level with `hostname` dimension as you already defined it but I have a doubt some lower aggregation like `".mean(by=['dbname','dbtype','hostname'])"`, is it not possible to not aggregate at all for them ?

File: modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+name: heartbeat
+
+transformation: false
+aggregation: ".mean(by=['hostname'])"
+
+exclude_not_running_vm: true
+
+signals:
+  signal:
+    metric: oracledb_Process_limits_value
```
[the remark this answers is not in the document]
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml:11]: I did some tests with "oracledb_up" and it was not working than expected : I receive alerts of heartbeat whereas all is ok on db side (up). Since I'm using process checks and it seems that it's working correctly.

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml [hunk 17]
```diff
@@ -0,0 +1,18 @@
+module: oracle
+name: "Process listener"
+id: listener
+
+transformation: true
+aggregation: ".mean(by=['hostname'])"
+
+signals:
+  signal:
+    metric: tnslsnr
```
[the remark this answers is not in the document]
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/01-listener.yaml:10]: No this is a real server process, which is not related to oracledb but directly on OS stack. But currently I don't know how implement additional process monitoring (I'm working with Jeremy on it)

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml (hunk 1 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/01-listener.yaml:18]: effectively, listener is related to DBs and works on a host. but oracle listener is to connect on a DB, not on a "DB host", that's why I prefer to keep the suggested tip.

File: modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml (hunk 2 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml:11]: Yes I can drop the aggrega and let the dimension manage at the source level. In fact, it os more readable to have a clear description in the incident such as host / db / dbtype, but basically all metrics could works with dimensions which are managed at the source level. 
=> I will drop aggegra for all metrics, it will be less "human readable" but will avoid any issues related to concurrent datapoint.

File: modules/prometheus-exporter_oracle/conf/05-alertlog.yaml (hunk 3 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/05-alertlog.yaml:5]: Yes, i will do and drop the aggregation.

File: modules/prometheus-exporter_oracle/conf/17-dbvagent.yaml (hunk 11 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/17-dbvagent.yaml:1]: it is related to a server process related to an oracle replication (standby db)
=> I will implement a tip. 
SOS DBA are aware about this, and will be fully able to manage the incidents related to.

File: modules/prometheus-exporter_oracle/conf/18-dbvnet.yaml (hunk 12 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/18-dbvnet.yaml:1]: same than above.

File: modules/prometheus-exporter_oracle/conf/19-dbvctl.yaml (hunk 13 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/19-dbvctl.yaml:1]: same than above

File: modules/prometheus-exporter_oracle/conf/readme.yaml (hunk 14 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/readme.yaml:1]: I will do a simple doc on github with this prometheus exporter. But oracle monitoring ref are exclusively internal from my side.

File: modules/prometheus-exporter_oracle/conf/readme.yaml (hunk 15 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/readme.yaml:2]: as this is related to CLARANET (currently) and for my team SOS DBA, I have not plan to doc this module for public access, and the deployment is related only for claranet and particular ecosystem that I have on SOS DBA (standards etc) . For example custom metrics are managed in SOS DBA tooling (onegit) and are related to our standards, and are not usable by others company.

=> maybe we could discuss quickly about this global subject (documentation), to provide me some explanations in order to be "compliant" from your side :)

File: modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml:11]: I am pretty sure the problems are due to your tests, you probably dropped some dimensions so it leads to alerts, this is the same for any metric but perhaps you stopped to change your dimensions after the switch to this one

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/01-listener.yaml:10]: if the metric does not come from the prometheus exporter oracledb it must not be in this module

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml (hunk 1 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/01-listener.yaml:18]: ok

File: modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml (hunk 2 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml:11]: no matter you aggregate or not, all dimensions are available in the alert title

File: modules/prometheus-exporter_oracle/conf/05-alertlog.yaml (hunk 3 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/05-alertlog.yaml:5]: ok but you still need to create the transformation variable

File: modules/prometheus-exporter_oracle/conf/readme.yaml (hunk 14 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/readme.yaml:1]: while all detectors on this module are usable with your "simple doc" no problem

File: modules/prometheus-exporter_oracle/conf/readme.yaml (hunk 15 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/readme.yaml:2]: feel free to join me on slack

File: modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml:11]: yes, perhaps a simple workshop together may help me as I'm not totally aware with spunk usage / hearbeat etc. I will ping Jeremy if he could help me on it.

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/01-listener.yaml:10]: hum yes but where it has to be implement ?  new dedicated module ?

File: modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml (hunk 2 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml:11]: yes, but it's a little non friendly and "clear" like below test : 

`Alert:
[Critical][sandbox-sosdba] Oracle alert.log errors count error(s) detected > 0 (0.0)

Cleared at:
2021-11-15T16:30Z

Dimensions:
{instance=localhost:9161, scheme=http, service.name=oracle-exporter-1, erreurora=ORA-00257, env=sandbox-sosdba, sfx_monitored=true, hostname=ora-batman, dbname=SFXCDB, port=9161, sf_metric=oracledb_AlertLogError_count, dbtype=container_DB, host.name=localhost, job=oracle-exporter-1}
`

independently I drop the aggregation :)

File: modules/prometheus-exporter_oracle/conf/05-alertlog.yaml (hunk 3 above)
Comment (mricois) [reply modules/prometheus-exporter_oracle/conf/05-alertlog.yaml:5]: ok , this means I have to add "aggregation: true" on it, and for all detectors ?

Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/05-alertlog.yaml:5]: except if you think that give the ability to user to change aggregation is risky regarding the detector behavior.

here I think it is safe the let the user customize this (for example to get an alert title with a subset of dimensions instead of all of them which could be a bit long as you noticed before)

File: modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml (hunk 2 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/02-dbisdown.yaml:11]: yes you get all dimensions if you do not aggregate which could be a bit long but at least it works in all cases and if you give the user a variable to modify the aggregation he still can aggregate himself depending on its own (custom) dimensions if he want more concise alert title

File: modules/prometheus-exporter_oracle/conf/01-listener.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/01-listener.yaml:10]: it depends on how you collect the metric to make this detector. I don't have enough info of its goal to answer but:
- either you use a new "source" of metrics and you have to create a new module
- or you can use an existing module which could cover your goal like https://github.com/claranet/terraform-signalfx-detectors/tree/master/modules/smart-agent_processes or https://github.com/claranet/terraform-signalfx-detectors/tree/master/modules/smart-agent_systemd-services

File: modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (xp-1000) [reply modules/prometheus-exporter_oracle/conf/00-heartbeat.yaml:11]: yes feel free to join me on huddle. what you need to understand is: as soon as you change dimensions on your agent configuration you can change generated MTS too.

especially, if you remove or rename a dimension (key or value) the "old" mts will stop to send datapoint and a new mts will be created.

the heartbeat will so trigger an alert for the disappearing mts and starts to monitor the new one.
forcing aggregation will mitigate this risk because if you aggregate on `host` for example, no matter your changes on others dimensions, while this dimensions does not change you will never get alert.
that said, in general we recommend to not use aggregation from module side except if really required and give the user a variable to define it himself depending on his environment.

File: modules/smart-agent_processes/conf/01-listener.yaml [hunk 18]
```diff
@@ -0,0 +1,18 @@
+module: oracle
```
Comment (xp-1000) [inline modules/smart-agent_processes/conf/01-listener.yaml:1]: the existing detector is generic, you must not touch to this module

**Review by xp-1000 — CHANGES_REQUESTED** [review]

File: modules/prometheus-exporter_oracledb/conf/00-heartbeat.yaml [hunk 19]
```diff
@@ -0,0 +1,14 @@
+module: oracle
+name: heartbeat
+
+transformation: true
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/conf/00-heartbeat.yaml:4]: you can remove this one, changing transformation has no effect on heartbeat so it is basically useless (but not risky neither)

File: modules/prometheus-exporter_oracledb/tags.tf [hunk 20]
```diff
@@ -0,0 +1,4 @@
+locals {
+  tags = ["prometheus-exporter", "oracle"]
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/tags.tf:2]:
```suggestion
  tags = ["prometheus-exporter", "oracledb"]
```

File: modules/smart-agent_processes/README.md [hunk 21]
```diff
@@ -103,6 +104,7 @@ the corresponding monitor configuration:
       - metricNames:
         - '*'
         - '!ps_count.processes'
+        - '!tnslsnr'
```
Comment (xp-1000) [inline modules/smart-agent_processes/README.md:107]: please run the make again (or git reset --hard this file) to cancel changes on it

File: modules/smart-agent_processes/detectors-gen.tf [hunk 22]
```diff
@@ -0,0 +1,25 @@
+resource "signalfx_detector" "listener" {
```
Comment (xp-1000) [inline modules/smart-agent_processes/detectors-gen.tf:1]: same here, to cancel

File: modules/smart-agent_processes/outputs.tf [hunk 23]
```diff
@@ -1,3 +1,8 @@
+output "listener" {
```
Comment (xp-1000) [inline modules/smart-agent_processes/outputs.tf:1]: to cancel

File: modules/smart-agent_processes/variables-gen.tf [hunk 24]
```diff
@@ -0,0 +1,57 @@
+# listener detector
```
Comment (xp-1000) [inline modules/smart-agent_processes/variables-gen.tf:1]: to cancel

**Review by xp-1000 — CHANGES_REQUESTED** [review]

File: modules/prometheus-exporter_oracledb/conf/readme.yaml [hunk 25]
```diff
@@ -0,0 +1,83 @@
+documentations:
+  - name: Prometheus Exporter for oracledb
+    url: https://github.com/iamseth/oracledb_exporter
+
+source_doc: |
+  The detectors of this module uses defaults metrics from the [oracledb exporter] (https://github.com/iamseth/oracledb_exporter) and optionnaly custom metrics from custom templates which have to be specified during the deployment.
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/conf/readme.yaml:6]:
```suggestion
  The detectors of this module uses defaults metrics from the [oracledb exporter](https://github.com/iamseth/oracledb_exporter).
```

File: modules/prometheus-exporter_oracledb/conf/readme.yaml [hunk 26]
```diff
@@ -0,0 +1,83 @@
+documentations:
+  - name: Prometheus Exporter for oracledb
+    url: https://github.com/iamseth/oracledb_exporter
+
+source_doc: |
+  The detectors of this module uses defaults metrics from the [oracledb exporter] (https://github.com/iamseth/oracledb_exporter) and optionnaly custom metrics from custom templates which have to be specified during the deployment.  
+  Check its documentation to install and configure it appropriately with your Oracle database host.
+
+  by default in this module, only the metric oracledb_up is used with the default.metrics related to the [oracledb exporter].
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/conf/readme.yaml:9]: to remove, it is not the default it is always true

File: modules/prometheus-exporter_oracledb/conf/readme.yaml [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+  by default in this module, only the metric oracledb_up is used with the default.metrics related to the [oracledb exporter]. 
+
+  ### Examples
+
+  Here is a sample configuration fragment for the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) agent using
+  the [prometheusexec receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusexecreceiver).
+  
+  In this example we have an oracle instance which is running in Multitenant mode, means that we have one Container Database and one pluggable database. 
+  
+  important : Both needs a dedicated prometheus port to upload metrics.
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/conf/readme.yaml:18]:
```suggestion
  _important_: Both needs a dedicated prometheus port to upload metrics.
```

File: modules/prometheus-exporter_oracledb/conf/readme.yaml [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+  
+  {PATH_TO_ORACLEDB_BIN} = directory related to the oracledb_exporter binary which is used,
+  {PATH_TO_ORACLEDB_TEMPLATE} = directory related to the templates which are used (default & custom)
+  {ORACLE_HOME}= Oracle Home related to rdbms software
+  {TNS_PORT} = Oracle listener port on which the DB is listening.
+  {CONTAINER_DB} = name of the container DB (instance name)
+  {PLUGGABLE_DB} = name of the pluggable Database
+  {HOSTNAME} = name of the DB server host
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/conf/readme.yaml:26]:
```suggestion
  Environment variables used in the following sample configuration:
    - `PATH_TO_ORACLEDB_BIN` = directory where is located the oracledb_exporter binary
    - `PATH_TO_ORACLEDB_TEMPLATE` = directory related to the templates which are used
    - `ORACLE_HOME` = Oracle Home related to rdbms software
    - `TNS_PORT` = Oracle listener port on which the DB is listening.
    - `CONTAINER_DB` = name of the container DB (instance name)
    - `PLUGGABLE_DB` = name of the pluggable Database
```

File: modules/prometheus-exporter_oracledb/conf/readme.yaml [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+  
+  ```yaml
```
Comment (xp-1000) [inline modules/prometheus-exporter_oracledb/conf/readme.yaml:28]: please append `$` at the start of each variable use and remove `{` `}` for what you did not define a variable (e.g. `{USER}` -> `myuser`)

**Review by xp-1000 — CHANGES_REQUESTED** [review]

File: modules/prometheus-exporter_oracledb/conf/00-heartbeat.yaml (hunk 19 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/conf/00-heartbeat.yaml:4]: okay, transformation is removed.

File: modules/prometheus-exporter_oracledb/tags.tf (hunk 20 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/tags.tf:2]: okay, changed, sorry.

File: modules/smart-agent_processes/README.md (hunk 21 above)
Comment (mricois) [reply modules/smart-agent_processes/README.md:107]: Done

File: modules/smart-agent_processes/variables-gen.tf (hunk 24 above)
Comment (mricois) [reply modules/smart-agent_processes/variables-gen.tf:1]: done

File: modules/smart-agent_processes/outputs.tf (hunk 23 above)
Comment (mricois) [reply modules/smart-agent_processes/outputs.tf:1]: done

File: modules/smart-agent_processes/detectors-gen.tf (hunk 22 above)
Comment (mricois) [reply modules/smart-agent_processes/detectors-gen.tf:1]: done

File: modules/prometheus-exporter_oracledb/conf/readme.yaml (hunk 25 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/conf/readme.yaml:6]: yes, done ..

File: modules/prometheus-exporter_oracledb/conf/readme.yaml (hunk 27 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/conf/readme.yaml:18]: done

File: modules/prometheus-exporter_oracledb/conf/readme.yaml (hunk 26 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/conf/readme.yaml:9]: done

File: modules/prometheus-exporter_oracledb/conf/readme.yaml (hunk 28 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/conf/readme.yaml:26]: done

File: modules/prometheus-exporter_oracledb/conf/readme.yaml (hunk 29 above)
Comment (mricois) [reply modules/prometheus-exporter_oracledb/conf/readme.yaml:28]: done

File: modules/prometheus-exporter_oracledb/conf/readme.yaml (hunk 25 above)
Comment (xp-1000) [reply modules/prometheus-exporter_oracledb/conf/readme.yaml]: not yet, commit the suggestion directly would preserve you of copy/paste error.

**Review by xp-1000 — APPROVED** [review]
Looks good to me, congrats!
